packages/protocol-kit/src/utils/on-chain-tracking/generateOnChainIdentifier.ts

```diff
@@ -0,0 +1,41 @@
+import { keccak256, toHex } from 'viem'
+
+/**
+ * Generates a hash from the given input string and truncates it to the specified size.
+ *
+ * @param {string} input - The input string to be hashed.
+ * @param {number} size - The number of bytes to take from the end of the hash.
+ * @returns {string} A hexadecimal string representation of the truncated hash, without the `0x` prefix.
+ */
+export function generateHash(input: string, size: number): string {
+  const fullHash = keccak256(toHex(input))
+  return toHex(fullHash.slice(-size)).replace('0x', '') // Take the last X bytes
+}
+
+/**
+ * Generates an on-chain identifier for tracking transactions on the blockchain.
+ * This identifier includes hased metadata such as the project name, platform, tool, and tool version.
+ *
+ * @param {string} project - The name of the project initiating the transaction.
+ * @param {string} [platform='Web'] - The platform from which the transaction originates (e.g., "Web", "Mobile", "Safe App", "Widget"...).
+ * @param {string} tool - The tool used to generate the transaction (e.g., "protocol-kit").
+ * @param {string} toolVersion - The version of the tool used to generate the transaction.
+ * @returns {string} A string representing the on-chain identifier, composed of multiple hashed segments.
+ */
+function generateOnChainIdentifier(
```

Comment:
Just a small suggestion. Could this function receive an object instead? It would allow to share more understandable code snippets in the docs. 
```
generateOnChainIdentifier({
  project: 'asdfasdf',
  // ...
})
```
better than
```
generateOnChainIdentifier('asdfasdf', ...)
```